### DISCUSSION TITLE
bundle config build in Gemfile, or additional explanation to console after bundle fails?

### DISCUSSION THREAD

**Discussion by jspraul** [ruby/rubygems#5954] [Q&A]
Can I set `bundle config build.tiny_tds` inside the Gemfile?

Setting the environment variable when running `bundle` works:
```
BUNDLE_BUILD__TINY_TDS=--with-freetds-include=/c/Ruby31-x64/msys64/ucrt64/include/freetds bundle
```
Setting the environment variable in the `Gemfile` does not work:
```
if RUBY_PLATFORM='x64-mingw-ucrt'
    ENV['BUNDLE_BUILD__TINY_TDS']='--with-freetds-include=/c/Ruby31-x64/msys64/ucrt64/include/freetds'
end
```
If this is not possible, is there a way to output additional help text after `An error occurred while installing tiny_tds` explaining exactly how to fix the issue?

**Comment by hsbt** [comment]
>Can I set bundle config build.tiny_tds inside the Gemfile?

You cannot set it from the Gemfile, because Bundler reads `BUNDLE_*` from the environment before evaluating it. Use `bundle config set --local build.tiny_tds ...` instead. 

>is there a way to output additional help text after An error occurred while installing tiny_tds explaining exactly how to fix the issue?

There is also no hook for a gem to add help text to a failed build, so that has to come from the gem's own extconf.rb output.
